.github/workflows/ci.yml

```diff
@@ -12,6 +12,7 @@ on:
 jobs:
   clang-format-check:
     runs-on: macos-latest
+    if: github.event.pull_request.fork == false || github.event_name == 'push'
     steps:
```

Comment:
I see `github.event.pull_request.head.repo.fork` being used instead of `github.event.pull_request.fork` in certain references. Can we add that to some of the builds in case `github.event.pull_request.fork` does not work we can hope the other does?
Ref: https://stackoverflow.com/a/70776678, https://github.com/orgs/community/discussions/25217#discussioncomment-3246904

Reply:
Both seem to be used interchangeably. Will test out and if it does not work, I will switch it to try again.